crates/cranelift/src/func_environ.rs

```diff
@@ -507,14 +524,23 @@ impl<'module_environment> FuncEnvironment<'module_environment> {
         let fuel = builder
             .ins()
             .load(ir::types::I64, ir::MemFlags::trusted(), addr, offset);
-        builder.def_var(self.fuel_var, fuel);
+
+        if self.tunables.outband_fuel {
```

Comment:
Is this function ever executed with `outband_fuel`? This seems like a confusing implementation of this function since the purpose of the pinned register is that it's always valid and doesn't need reloading.